### PR TITLE
feat: updates to match the latest kona tag

### PR DESCRIPTION
**Pull request by leruaa**
Many fixes to comply to updates on kona `kona-client-v0.1.0-beta.1` tag

* Dependencies updates
* Traits refactoring
* Replace `anyhow::Error` by `OracleProviderError`

### CODE REVIEW

File: utils/client/src/oracle/mod.rs
Comment (ratankaliani) [inline utils/client/src/oracle/mod.rs:1]: Why did we change the cache to a Mutex here?

File: utils/client/src/oracle/mod.rs [hunk 1]
```diff
@@ -75,6 +86,10 @@ impl HintWriterClient for InMemoryOracle {
     }
 }
 
+impl FlushableCache for InMemoryOracle {
+    fn flush(&self) {}
```
Comment (leruaa) [inline utils/client/src/oracle/mod.rs:90]: Oooops

File: utils/client/src/oracle/mod.rs
Comment (leruaa) [reply utils/client/src/oracle/mod.rs:1]: Needed to implement FlushableCache for InMemoryOracle

File: utils/build/src/lib.rs
Comment (ratankaliani) [inline utils/build/src/lib.rs:1]: Can you remove the diff from this file (add the space back)

File: utils/client/src/driver.rs [hunk 2]
```diff
@@ -173,13 +176,14 @@ impl<O: CommsClient + Send + Sync + Debug> MultiBlockDerivationDriver<O> {
...
                             self.pipeline
-                                .signal(Signal::Reset {
+                                .signal(Signal::Reset(ResetSignal {
                                     l2_safe_head: self.l2_safe_head,
                                     l1_origin: self
                                         .pipeline
                                         .origin()
                                         .ok_or_else(|| anyhow!("Missing L1 origin"))?,
-                                })
+                                    system_config: None,
```
Comment (leruaa) [inline utils/client/src/driver.rs:185]: @ratankaliani the field `system_config` is new, not sure if it's right to set to None

**Review by ratankaliani — APPROVED** [review]
Looks good! We're now Holocene-compat 🚀

File: utils/client/src/driver.rs (hunk 2 above)
Comment (ratankaliani) [reply utils/client/src/driver.rs:185]: Relevant fn: https://github.com/anton-rs/kona/blob/cddeaf4e229c0fc5449c146efe0b82bfc614b267/crates/driver/src/pipeline.rs#L25-L99

**Pull request merged**